<a href="https://colab.research.google.com/github/aliccialc/HackerRank-Python-R/blob/main/HackerRank.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

1. Given an integer, n, perform the following conditional actions: 

  If n is odd, print Weird

  If n is even and in the inclusive range of 2 to 5, print Not Weird

  If n is even and in the inclusive range of 6 to 20, print Weird

  If n is even and greater than 20, print Not Weird

In [ ]:
import math
import os
import random
import re
import sys



if __name__ == '__main__':
    n = int(input().strip())
    if n % 2 == 1:
        print("Weird")
    elif n % 2 == 0 and n>=2 and n<=5:
        print("Not Weird")
    elif n % 2 == 0 and n>=6 and n<=20:
        print("Weird")
    elif n % 2 == 0 and n>20:
        print("Not Weird")

8
Weird


2. The provided code stub reads two integers from STDIN, a and b. 

  Add code to print three lines where:

  The first line contains the sum of the two numbers.

  The second line contains the difference of the two numbers (first - second).
  
  The third line contains the product of the two numbers.


In [ ]:
if __name__ == '__main__':
    a = int(input())
    b = int(input())
    print(a + b)
    print(a - b)
    print(a * b)

3
5
8
-2
15


3. The provided code stub reads two integers, a and b, from STDIN.

  Add logic to print two lines. The first line should contain the result of integer division,  a// b. The second line should contain the result of float division, a / b.

  No rounding or formatting is necessary.

In [ ]:
if __name__ == '__main__':
    a = int(input())
    b = int(input())
    print(a // b)
    print(a / b)


123
3
41
41.0


4. The provided code stub reads and integer, n, from STDIN. For all non-negative integers i < n, print i^2.

In [ ]:
if __name__ == '__main__':
    n = int(input())
    for i in range(0, n):
     print(i*i)
      
    
    

5
0
1
4
9
16


5. Given a year, determine whether it is a leap year. If it is a leap year, return the Boolean True, otherwise return False.

  Note that the code stub provided reads from STDIN and passes arguments to the is_leap function. It is only necessary to complete the is_leap function.

In [ ]:
def is_leap(year):
    leap = False
    if year % 4 == 0:
        leap = True
    if year % 100 == 0:
        if year % 400 == 0:
            leap = True
        else:
            leap = False  
    return leap

year = int(input())
print(is_leap(year))

1990
False


6. The included code stub will read an integer, n, from STDIN. Without using any string methods, try to print the following:

  123...n

  Note that "..." represents the consecutive values in between.

In [ ]:
if __name__ == '__main__':
    n = int(input())
    
    for i in range(1, n + 1):
      print(i, end='')

3
123